server/src/main/java/org/elasticsearch/search/aggregations/bucket/composite/CompositeValuesCollectorQueue.java

```diff
@@ -26,15 +26,17 @@
 import org.elasticsearch.common.util.BigArrays;
 import org.elasticsearch.common.util.IntArray;
 import org.elasticsearch.search.aggregations.LeafBucketCollector;
+import org.elasticsearch.search.aggregations.support.BreakingPriorityQueueWrapper;
 
 import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
+import java.util.function.Consumer;
 
 /**
  * A specialized {@link PriorityQueue} implementation for composite buckets.
  */
-final class CompositeValuesCollectorQueue extends PriorityQueue<Integer> implements Releasable {
+final class CompositeValuesCollectorQueue extends BreakingPriorityQueueWrapper<Integer> implements Releasable {
```

Comment:
I wonder that we should extend close to make sure it calls `super.close()` which doesn't seem to be the case here?

Reply:
Whoops, yes, good catch!